zipline/data/us_equity_pricing.py

```diff
@@ -205,6 +269,15 @@ def _write_internal(self, filename, calendar, iterator):
         }
 
         earliest_date = None
+        calendar = self._calendar
+
+        if assets is not None:
+            @apply
```

Comment:
in the other branch iterator is already the iterator, here I just wanted to make another wrapper around that if we are passed assets.